adapter/memQuota/rollingWindow.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memQuota
+
+// Implements a rolling window that allows N units to be allocated per rolling time interval.
+// Time is abstracted in terms of ticks, provided by the caller, decoupling the
+// implementation from real-time, enabling much easier testing and more flexibility.
+type rollingWindow struct {
```

Comment:
We're talking about 1 26 line function. What's there now is tested, works, has all the semantics we need, and is very efficient. I'd like to leave as-is.